src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -367,23 +390,50 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
             }
         }
 
-        internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestMessage request)
+        internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestMessage request, bool stripAuthorization)
         {
             if (client == null) { throw new ArgumentNullException("client"); }
             if (request == null) { throw new ArgumentNullException("request"); }
 
             _cancelToken = new CancellationTokenSource();
-            return client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, _cancelToken.Token).GetAwaiter().GetResult();
+            HttpResponseMessage response = client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, _cancelToken.Token).GetAwaiter().GetResult();
+
+            if (
+                System.Net.HttpStatusCode.Found == response.StatusCode
+                &&
+                stripAuthorization 
+                )
+            {
+                _cancelToken.Cancel();
```

Comment:
> _cancelToken.Cancel(); [](start = 15, length = 23)

I am a little confused by this logic.  It looks like an asynchronous call is being cancelled here.  Is there a possibility of a race condition?  Can you add a comment indicating what is being cancelled?